### PR TITLE
users: fix displayed accounts on debian

### LINKED ISSUE #3166: User "nobody" is shown in Accounts panel under Debian
Under Debian, the user "nobody" is shown in the Accounts panel.

Fedora:
```
# getent passwd nobody
nobody:x:99:99:Nobody:/:/sbin/nologin
```
Debian
```
# getent passwd nobody
nobody:x:65534:65534:nobody:/nonexistent:/usr/sbin/nologin
```

### PR DESCRIPTION
**Pull request by larsu**
Closes #3166

### CODE REVIEW

File: pkg/users/local.js [hunk 1]
```diff
@@ -206,7 +206,7 @@ function parse_passwd_content(content) {
             password: column[1],
             uid: parseInt(column[2], 10),
             gid: parseInt(column[3], 10),
-            gecos: column[4],
+            gecos: column[4].split(',')[0],
```
Comment (stefwalter) [inline pkg/users/local.js]: I think we need to strip trailing commas here. It's reasonable for someone to have commas in their full name, for example.

File: src/bridge/cockpitdbususer.c [hunk 2]
```diff
@@ -48,11 +49,16 @@ populate_passwd_props (GHashTable *props,
     }
   else
     {
+      gchar *full;
+
+      full = g_strndup (pw->pw_gecos, strcspn (pw->pw_gecos, ","));
```
Comment (stefwalter) [inline src/bridge/cockpitdbususer.c]: Again, I think that stripping trailing commas would make more sense.

**Comment by larsu** [comment]
I guess that makes sense. Will add it in a bit.

Note though that neither chfn(1) nor accountsservice allows commas (even escaped) in one of the fields.

**Comment by stefwalter** [comment]
The cockpit-session stuff is no longer used. I'm removing it in #3897. It's worth either rebasing on that, or removing the changes to session.c here.

**Comment by larsu** [comment]
Thanks!

**Comment by stefwalter** [comment]
Dropped the session.c changes while merging. In theory could have waited for #3897 ... but adding code to session.c (priveleged setuid process) is risky right before a release.

**Pull request closed without merging**